packages/base/src/Base.ts

```diff
@@ -78,6 +79,27 @@ export function SpectrumMixin<T extends Constructor<UpdatingElement>>(
             return this.dir === 'ltr';
         }
 
+        public hasVisibleFocusInTree(): boolean {
+            const activeElement = (this.getRootNode() as Document)
+                .activeElement as HTMLElement;
+            if (!activeElement) {
+                return false;
+            }
+            let hasVisibleFocusInTree = false;
+            try {
+                // Browsers without support for the `:focus-visible`
+                // selector will throw on the following test (Safari, older things).
+                // Some won't throw, but will be focusing item rather than the menu and
+                // will rely on the polyfill to know whether focus is "visible" or not.
+                hasVisibleFocusInTree =
+                    activeElement.matches(':focus-visible') ||
+                    activeElement.matches('.focus-visible');
+            } catch (error) {
+                hasVisibleFocusInTree = activeElement.matches('.focus-visible');
+            }
+            return hasVisibleFocusInTree;
+        }
+
```

Comment:
Nice! I like that outside of tests, this PR deletes more code than it adds, by consolidating this shared functionality.

Is there a benefit to storing the temporary result in a variable vs. returning from whichever case?

```suggestion
        public hasVisibleFocusInTree(): boolean {
            const activeElement = (this.getRootNode() as Document)
                .activeElement as HTMLElement;
            if (!activeElement) {
                return false;
            }
            // Browsers without support for the `:focus-visible`
            // selector will throw on the following test (Safari, older things).
            // Some won't throw, but will be focusing item rather than the menu and
            // will rely on the polyfill to know whether focus is "visible" or not.
            try {
                return activeElement.matches(':focus-visible') || activeElement.matches('.focus-visible');
            } catch (error) {
                return activeElement.matches('.focus-visible');
            }
        }
```